Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -23,6 +23,7 @@ extension SyntaxProtocol {
   /// Expand all uses of the given set of macros within this syntax node.
   public func expand(
     macros: [String: Macro.Type],
+    conformsTo conformanceMap: [String: InheritedTypeListSyntax],
```

Comment:
Since `SyntaxProtocol.expand` is a public function, adding a new parameter to it is a source-breaking change. But I think we can safely default it to an empty dictionary for API compatibility.

Reply:
If we would go with `MacroSpec` in the `assertMacroExpansion` function, shouldn't we also at it here?

Reply:
Yes, this function probably changes shape completely with the introduction of `MacroSpec`.

Reply:
@ahoppen added `MacroSpec` here as well, while keeping the existing definition for compatibility.